src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> | object {
```

Comment:
The return type `GroupsMap<T> | object` could be more specific. Since `GroupsMap<T>` is already an object, specifying `| object` is redundant and could be misleading. It's better to just return `GroupsMap<T>`.